app/models/spree/product_decorator.rb

```diff
@@ -84,6 +84,12 @@
       select('distinct spree_products.*')
   }
 
+  scope :in_distributors, lambda { |distributors|
```

Comment:
but `#in_distributors` was used already. Was it a method instead of a scope? Won't we have two conflicting definitions?

Reply:
There's an `in_distributor` scope, but it only takes a single object. This one takes a relation, with multiple distributors.

Reply:
Isnt this copy-paste-free version better?
```ruby
  # Products distributed via the given distributor through an OC
  scope :in_distributor, lambda { |distributor|
    distributor = distributor.respond_to?(:id) ? distributor.id : distributor.to_i

    in_distributor_base(distributor).
      select('distinct spree_products.*')
  }

  scope :in_distributors, lambda { |distributors|
    in_distributor_base(distributors).uniq
  }

  scope :in_distributor_base, lambda { |distributors|
    with_order_cycles_outer.
      where('(o_exchanges.incoming = ? AND o_exchanges.receiver_id IN (?))', false, distributors)
  }

```